libappstream-builder/asb-context.c

```diff
@@ -565,12 +565,14 @@ asb_context_setup (AsbContext *ctx, GError **error)
 	/* icons is nuked; we can re-decompress from the -icons.tar.gz */
 	if (!asb_utils_ensure_exists (priv->icons_dir, error))
 		return FALSE;
+
+	g_autofree gchar *icons_dir_lodpi = NULL;
+	icons_dir_lodpi = g_build_filename (priv->icons_dir, "64x64", NULL);
+	if (!asb_utils_ensure_exists (icons_dir_lodpi, error))
+		return FALSE;
+
 	if (priv->flags & ASB_CONTEXT_FLAG_HIDPI_ICONS) {
 		g_autofree gchar *icons_dir_hidpi = NULL;
-		g_autofree gchar *icons_dir_lodpi = NULL;
-		icons_dir_lodpi = g_build_filename (priv->icons_dir, "64x64", NULL);
-		if (!asb_utils_ensure_exists (icons_dir_lodpi, error))
-			return FALSE;
```

Comment:
doesn't this change the logic? i.e. we'd put the icons in priv->icons_dir -- unless we used HIDPI and then we'd put them in priv->icons_dir/64x64 and priv->icons_dir/128x128

Reply:
I'm wondering if we should just remove enable-hidpi completely and just always write 64x64 and 128x128 icons. It would certainly make the code a lot simpler...

Reply:
> I'm wondering if we should just remove enable-hidpi completely and just always write 64x64 and 128x128 icons. It would certainly make the code a lot simpler...

Aggreed.

Reply:
Shouldn't the 64x64px icons always be available anyway? That size is pretty much the only size required to be present by the specification, with all other icon dimensions being optional.

Reply:
Dude, this is for super old clients that expected the icons in /usr/share/app-info/icons/$origin/ i.e. without the 64x64 suffix. I think we can drop support for that now.